### DISCUSSION TITLE
Formik with validationSchema and initial empty values?

### DISCUSSION THREAD

**Discussion by FezVrasta** [jaredpalmer/formik#3981] [General]
I would like to have my form start with empty form fields, but require to pass a validation where the fields are non-empty.

```ts
const schema = Yup.object().shape({
  price: Yup.number().required()
});

useFormik<InferType<typeof schema>>({
  validationSchema: schema,
  initialValues: {
    price: undefined, // This errors because it's not a number
  },
  onSubmit(value) {
    // handle submit
  }
});
```

If I `Partial<>` the `useFormik` generic I then get an `onSubmit` value that doesn't match with my schema validator.

I think it's totally reasonable to provide empty input fields when a form is presented to a user, and expect them to fill them with the correct value. How can I do this with Formik, Yup, and TypeScript?

**Comment by runoncedev** [comment]
I have a similar problem, did you find any workarounds?

**Reply by FezVrasta** [reply]
Not really 🤷‍♂️
